.github/workflows/ci.yml

```diff
@@ -0,0 +1,99 @@
+name: CI
+
+env:
+  go-version: '1.14.6'
+  GCLOUD_KEY: ${{ secrets.GCLOUD_KEY }}
+  PROJECT_NAME: ${{ secrets.PROJECT_NAME }}
+  CLUSTER_NAME: ${{ secrets.CLUSTER_NAME }}
+  CLUSTER_ZONE: ${{ secrets.CLUSTER_ZONE }}
+  ES_PASSWD: ${{ secrets.ES_PASSWD }}
+
+# Trigger the workflow on all pull requests, and on push to specific branches
+on:
+  pull_request:
+    # Don't trigger tests if only updating docs
+    paths-ignore:
+      - '**.md'
+  push:
+    branches:
+      - staging
+      - trying
+    # Don't trigger tests if only updating docs
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  # always run the unit tests, and run them first, as they are relatively cheap
+  unittests:
+    runs-on: ubuntu-latest
+    steps:
+     - name: checkout
+       uses: actions/checkout@v2
+     - name: set up go
+       uses: actions/setup-go@v2
+       with:
+         go-version: ${{ env.go-version }}
+     - name: setup env
+       run: make
+     - name: fmt, tidy, lint
+       run: |
+         make genproto
+         make test-tidy
+         make test-fmt
+         make lint
+     - name: unit tests (except app test)
+       run: make test-no-app-test
+     - name: app test
+       run: make test-only-app-test
```

Comment:
So these run sequentially? This will double the time it takes to run them.

While `fmt, tidy, lint` is quick, the other two sets of tests take 9-10 minutes each -- that's why I've split them and made them run in parallel.

Travis lets us run a set of tests and once they all passed (in parallel) -- start the next set. Is this possible here?

A bigger problem is that we must push to DockerHub prior to running the system tests. The system tests use the DockerHub image to run.

Reply:
Good catches. I made these changes. The only difference compared to the current travis config is that, to keep the workflow simpler, the dockerpush task that runs only on a push to `develop` is in a separate workflow file, with no preconditions (since, as discussed, bors only allows a push to this branch once all tests have passed). I think this is more readable than smushing it into the main CI workflow. Let me know if you disagree.